source/_posts/Docker-Swarm.md

```diff
@@ -0,0 +1,240 @@
+```
+
+
+Zie hier een voorbeeld Docker-compose file om een Portainer container en een Portainer agent te deployen als stack** op Docker Swarm.
+
+``` yaml
+version: '3.7'
+
+services:
+  agent:
+    image: portainer/agent
+    environment:
+      AGENT_CLUSTER_ADDR: tasks.agent
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /var/lib/docker/volumes:/var/lib/docker/volumes
+    networks:
+      - private
+    deploy:
+      mode: global
+  portainer:
+    image: portainer/portainer
+    command: -H tcp://tasks.agent:9001 --tlsskipverify
+    volumes:
+      - portainer-data:/data
+    networks:
+      - private
+      - public
+    deploy:
+      placement:
+        constraints:
+          - node.role == manager
+      labels:
+        - traefik.frontend.rule=Host:portainer.${DOMAIN}
+        - traefik.enable=true
+        - traefik.port=9000
+        - traefik.tags=public
+        - traefik.docker.network=public
+        - traefik.redirectorservice.frontend.entryPoints=http
+        - traefik.redirectorservice.frontend.redirect.entryPoint=https
+        - traefik.webservice.frontend.entryPoints=https
+
+networks:
+  private:
+    driver: overlay
+    name: private
+  public:
+    external: true
+
+volumes:
+  portainer-data: {}
+
+```
+Een kleine samenvatting wat er gebeurd in dit Docker-compose bestand:
+
+- We maken een container aan op basis van portainer/portainer en portainer/agent.
+- We publiceren poort 9000 voor de UI.
+- We regelen de Traefik configuratie in.
+- We mounten een volume voor de portainer data.
+- We zorgen dat de Portainer container wordt uitgerold op een manager in het cluster en dat de agent op elke node in het cluster wordt uitgerold. 
+- We maken een netwerk aan met de naam Private en koppelen dit aan de agent en de Portainer UI.
+- We koppelen de Portainer UI ook aan het netwerk Public wat we hebben aangemaakt in de Traefik deploy.
+
+Om de stack** uit te rollen voeren we het volgende commando uit om een manager node: 
+```
+docker stack deploy -c docker-compose.portainer.yml portainer
+```
+
+Vanaf nu kunnen we Portainer benaderen op de URL https://portainer.yourdomain.com en kunnen we hier vandaan de rest van de services\* en stacks\*\* deployen en beheren.
+
+
+Conclusie
+---
+Ik vind dat de combinatie tussen de verschillende tools een hele goede basis zijn als applicatie landschap. Het is flexibel, schaalbaar en niet bijzonder complex. Docker Swarm staat ook bekend om zijn eenvoudigheid. Aangezien ik geen infrastructuur met honderden nodes en duizenden containers hoef te hosten is dit een hele geschikte oplossing. Docker Swarm heeft niet een hele steile leercurve dus je kunt er al snel mee aan de slag. Door de reverse proxy van Traefik kunnen virtuele hosts automatisch worden aangemaakt en is SSL meteen geregeld. Met Portainer als UI voor het beheer kun je in een mum van tijd een heel applicatie landschap optuigen zonder al te veel tijd en kosten te moeten investeren.
```

Comment:
Ik vind dat de combinatie van de verschillende tools een goede basis is voor het applicatielandschap.

Reply:
Wat kost dit? Je was toch een Hollander